src/Components/v2/ArtworkFilter/ArtworkFilterContext.tsx

```diff
@@ -0,0 +1,184 @@
+  acquireable?: boolean
+  at_auction?: boolean
+  color?: string
+  for_sale?: boolean
+  height: string
+  inquireable_only?: boolean
+  keyword?: string
+  major_periods: string[]
+  medium?: string
+  offerable?: boolean
+  page: number
+  partner_id?: string
+  price_range: string
+  sort: string
+  width: string
+}
+
+export const initialFilterState = {
+  height: "*-*",
+  major_periods: [],
+  page: 1,
+  price_range: "*-*",
+  sort: "-decayed_merch",
+  width: "*-*",
+}
+
+export const FilterContext = React.createContext<{
+  filters: Filters
+  hasFilters: boolean
+  isDefaultValue: (name: string) => boolean
+  rangeToTuple: (name: string) => [number, number]
+  resetFilters: () => void
+  setFilter: (name: string, value: any) => void
+  unsetFilter: (name: string) => void
+}>({
+  filters: initialFilterState,
+  hasFilters: false,
+  isDefaultValue: null,
+  rangeToTuple: null,
+  resetFilters: null,
+  setFilter: null,
+  unsetFilter: null,
+})
+
+export const FilterContextProvider: React.FC<
+  Filters & { children: React.ReactNode }
+> = ({
+  children,
+  height = "*-*",
+  major_periods = [],
+  medium = "",
+  page = 1,
+  price_range = "*-*",
+  sort = "-decayed_merch",
+  width = "*-*",
+  ...rest
+}) => {
+  const [state, dispatch] = useReducer(filterReducer, {
+    height,
+    major_periods,
+    medium,
+    page,
+    price_range,
+    sort,
+    width,
+    ...rest,
+  })
+  const isMounted = useDidMount()
+  // const updateOnChange = Object.keys(state).map(prop => state[prop])
+
+  useEffect(() => {
+    if (isMounted) {
+      const queryString = urlFragmentFromState(state)
+      window.history.pushState(
+        {},
+        null,
+        `${window.location.pathname}?${queryString}`
+      )
+    }
+  })
+
+  /**
+   * TODO: use this to trigger resetting of fields when back/fwd buttons are hit
+   * because otherwise it doesn't always happen.
+   */
+  useEffect(() => {
+    window.addEventListener("popstate", handleChange)
+    return () => window.removeEventListener("popstate", handleChange)
+  }, [])
+
+  function handleChange() {
+    console.log("derive state from URL here, and update!")
+  }
+
+  const value = {
+    filters: state,
+    hasFilters: hasFilters(state),
+
+    isDefaultValue: field => {
+      return isDefaultFilter(field, state[field])
+    },
+
+    rangeToTuple: range => {
+      return rangeToTuple(state, range)
+    },
+
+    setFilter: (name, val) => {
+      dispatch({
+        type: "SET",
+        payload: {
+          name,
+          value: val,
+        },
+      })
+    },
+
+    unsetFilter: name => {
+      dispatch({
+        type: "UNSET",
+        payload: {
+          name,
+        },
+      })
+    },
+
+    resetFilters: () => {
+      dispatch({
+        type: "RESET",
+        payload: null,
+      })
+    },
+  }
+
+  return (
+    <FilterContext.Provider value={value}>{children}</FilterContext.Provider>
+  )
+}
+
+const filterReducer = (state, action) => {
```

Comment:
I removed some of the types around this because React's type definitions are very, very good and can infer all the way from the context, with autocomplete.